ansible/devutil/devices/ansible_hosts.py

```diff
@@ -502,16 +506,41 @@ def _check_results(self, caller_info, module_info, results, module_ignore_errors
         if isinstance(self, AnsibleHosts):
             hosts_str = json.dumps(self.hostnames)
         elif isinstance(self, AnsibleHost):
-            hosts_str = json.dumps(self.hostnames[0])
-            results = results.get(self.hostnames[0], {})
+            hosts_str = self.hostnames[0]
         else:
             raise TypeError("Unsupported type of object: {}".format(type(self)))
 
-        if isinstance(module_info, dict):
+        unreachable = False
+        failed = False
+        for _, module_results in results.items():
+            if isinstance(module_results, list):
+                for result in module_results:
+                    if "reachable" in result and not result["reachable"]:
+                        unreachable = True
+                    if "unreachable" in result and result["unreachable"]:
+                        unreachable = True
+                    if "failed" in result and result["failed"]:
+                        failed = True
+                    if unreachable and failed:
+                        break
+            elif isinstance(module_results, dict):
+                if "reachable" in module_results and not module_results["reachable"]:
+                    unreachable = True
+                if "unreachable" in module_results and module_results["unreachable"]:
+                    unreachable = True
+                if "failed" in module_results and module_results["failed"]:
+                    failed = True
```

Comment:
Same as line518-523. Extract a method such as `check_failure()->unreachable, failed:` may be better?

Reply:
But this code is too simple; writing it twice is also good.

Reply:
This code is to conditionally update variables `unreachable` and `failed` within the scope. Replace the duplication with a common method would be a little bit tricky. It may be better to keep the duplication to keep it simple.

Reply:
Yes, if did it might be like `unreachable, failed=check_failure()` to update outside scope var.

Reply:
`unreachable, failed=check_failure()` is not going to work. Function `check_failure()` always returns either True or False. Then variables `unreachable` and `failed` might be flipped between True and False. Each call of `check_failure()` could flip them to False. Then the logic is not the same as the original code. In the original code, we only flip the variables to True when some conditions are met. Otherwise, **no operation**.

Reply:
Oh, that makes sense. Thank you for the explanation!